src/components/Options/index.tsx

```diff
@@ -81,10 +81,15 @@ export default class Options extends Component<Props, State> {
   onPreprocessorEnabledChange(event: Event) {
     const el = event.currentTarget as HTMLInputElement;
 
-    const preprocessorState = this.props.preprocessorState;
-    const preprocessor = el.name.split('.')[0] as keyof typeof preprocessorState;
-    preprocessorState[preprocessor].enabled = el.checked;
-    this.props.onPreprocessorOptionsChange(preprocessorState);
+    const preprocessor = el.name.split('.')[0] as keyof PreprocessorState;
+
+    this.props.onPreprocessorOptionsChange({
```

Comment:
Forgot to mention - I wrote a utility for doing the immutable deep set stuff, but then found a good one already available. Think we should pull it in? It's 180b and we have this pattern in a couple spots:

https://github.com/fwilkerson/clean-set

Seems like it might help with the typings as well.

Reply:
This looks really good. Yeah, we should use it.

In this case, we should probably just be getting all the data from the form fresh each time (this is how I do it with the webp encoder).